tests/fixture/mod.rs

```diff
@@ -74,30 +70,25 @@ impl Fixture {
         );
     }
 
-    pub fn create_wrangler_toml(&self, content: &str) {
-        let content = &format!(
-            r#"
-            name = "test"
-            workers_dev = true
-            {}
-        "#,
-            content
-        );
-        self.create_file("wrangler.toml", content);
+    pub fn create_wrangler_toml(&self, wrangler_toml: WranglerToml) {
+        self.create_file("wrangler.toml", &toml::to_string(&wrangler_toml).unwrap());
     }
 
-    pub fn cleanup(&self) {
-        let path = self.get_path();
-        assert!(path.exists(), format!("{:?} does not exist", path));
-
-        // Workaround https://github.com/rust-lang/rust/issues/29497
-        if cfg!(target_os = "windows") {
-            let mut command = Command::new("cmd");
-            command.arg("rmdir");
-            command.arg("/s");
-            command.arg(&path);
-        } else {
-            fs::remove_dir_all(&path).unwrap();
+    pub fn lock(&self) -> MutexGuard<'static, ()> {
+        use std::sync::Mutex;
+
+        lazy_static! {
+            static ref ONE_TEST_AT_A_TIME: Mutex<()> = Mutex::new(());
+        }
+
+        ONE_TEST_AT_A_TIME.lock().unwrap_or_else(|e| e.into_inner())
+    }
+}
+
+impl Drop for Fixture {
```

Comment:
our impl specifies that if the test succeeds (the test thread does not panic) we drop the temp dir. otherwise it is left in place. the `eprintln!` written out during `Fixture::new()` tells you where to find your failed test assets.

Reply:
omg wow. yes. one million times yes.